### ISSUE TITLE
PSR12.Files.FileHeader not an accurate definition of the file header

### ISSUE THREAD

**Issue by rodion1322** [squizlabs/PHP_CodeSniffer#3596]
**Describe the bug**
Skipped checking opened tag in header in file containing only php with misprint

**Code sample**
```php
'<?php

return [
    'property1' => 'value',
    'property2' => 'value',
];

```

**Custom ruleset**
N/A

**To reproduce**
Just any symbol/content before opened tag and code don't has another php construction like "use" and etc

**Expected behavior**
Expected a **"The file header must be the first content in the file"**.
Also i seeing comment in sniff about impossible identify proper header.
But reading **PSR12 header 2.2 "The closing ?> tag MUST be omitted from files containing only PHP."**. it can be concluded that files containing only php, should not contain a closing tag and can be accurately identified
And files with multiple tags must have end tags as they are actually mixed

**Versions (please complete the following information):**
 - OS: MacOS 11.6
 - PHP: 7.4
 - PHPCS: master
 - Standard: PSR12

**Comment by gsherwood** [comment]
If you put something in the file that is included in a file header, you'll get the error message correctly:
```
`<?php

namespace Vendor\Package;

return [
    'property1' => 'value',
    'property2' => 'value',
];
```
The sample code you provided doesn't include any of the file header components as defined by PSR12 so the sniff doesn't enforce a header.

**Comment by rodion1322** [comment]
I may be wrong, but thanks anyway.

**Comment by jrfnl** [comment]
@rodion1322 You may be interested in the `Generic.PHP.CharacterBeforePHPOpeningTag` sniff. That should detect the character before the open tag and flag it.

**Comment by rodion1322** [comment]
@jrfnl Great, it is solve my problem. Thanks!
